vmdb/app/controllers/api_controller/action.rb

```diff
@@ -0,0 +1,33 @@
+class ApiController
+  module Action
+    private
+
+    def api_action(type, id, options = {})
+      cspec = collection_config[type]
+      klass = cspec[:klass].constantize
+
+      result = yield(klass) if block_given?
+
+      add_href_to_result(result, type, id) if options[:add_href]
+      log_result(result)
+      result
+    end
+
+    def queue_api_action(object, summary, options)
```

Comment:
I think the method name is confusing only because you have another method called api_action up above which is not related.  Typically the sync/async versions of a method are named foo/foo_queue.  But in this case, api_action is boilerplate stuff, and queue_api_action actually calls the real action.  So, the names together make it confusing.

Not sure of a better name.  I don't think api makes sense in the name, because at this point you're past the REST API and calling object methods.  Maybe `queue_object_action` or `queue_operation`?  @jrafanie You're good at names :stuck_out_tongue_winking_eye: